test/e2e/airgapped/airgapped_test.go

```diff
@@ -0,0 +1,212 @@
+	. "github.com/onsi/gomega"
+
+	"github.com/vmware-tanzu/tanzu-cli/pkg/constants"
+	"github.com/vmware-tanzu/tanzu-cli/test/e2e/framework"
+	pluginlifecyclee2e "github.com/vmware-tanzu/tanzu-cli/test/e2e/plugin_lifecycle"
+)
+
+var _ = framework.CLICoreDescribe("[Tests:E2E][Feature:Airgapped-Plugin-DownloadBundle-UploadBundle-Lifecycle]", func() {
+
+	Context("Download plugin bundle, Upload plugin bundle and plugin lifecycle tests with plugin group 'vmware-tkg/v0.0.1'", func() {
+		// Test case: download plugin bundle for plugin-group vmware-tkg/v0.0.1
+		It("download plugin bundle with specific plugin-group vmware-tkg/v0.0.1", func() {
+			err := tf.PluginCmd.DownloadPluginBundle(e2eTestLocalCentralRepoImage, []string{"vmware-tkg/v0.0.1"}, filepath.Join(tempDir, "plugin_bundle_vmware-tkg-v0.0.1.tar.gz"))
+			Expect(err).To(BeNil(), "should not get any error while downloading plugin bundle with specific group")
+		})
+
+		// Test case: upload plugin bundle downloaded using vmware-tkg/v0.0.1 plugin-group to the airgapped repository
+		It("upload plugin bundle that was downloaded using vmware-tkg/v0.0.1 plugin-group to the airgapped repository", func() {
+			err := tf.PluginCmd.UploadPluginBundle(e2eAirgappedCentralRepo, filepath.Join(tempDir, "plugin_bundle_vmware-tkg-v0.0.1.tar.gz"))
+			Expect(err).To(BeNil(), "should not get any error while uploading plugin bundle")
+		})
+
+		// Test case: validate that the updating the discovery source to point to new airgapped repository works
+		It("update discovery source to point to new airgapped repository discovery image", func() {
+			_, err := tf.PluginCmd.UpdatePluginDiscoverySource(&framework.DiscoveryOptions{Name: "default", SourceType: framework.SourceType, URI: e2eAirgappedCentralRepoImage})
+			Expect(err).To(BeNil(), "should not get any error for plugin source update")
+			// Also set the skip signature verification for the airgapped repository discovery image
+			os.Setenv(constants.PluginDiscoveryImageSignatureVerificationSkipList, e2eAirgappedCentralRepoImage)
+		})
+
+		// Test case: Validate that the correct plugins and plugin group exists with `tanzu plugin search` and `tanzu plugin group search` output
+		It("validate the plugins from group 'vmware-tkg/v0.0.1' exists", func() {
+			// search plugin groups
+			pluginGroups = pluginlifecyclee2e.SearchAllPluginGroups(tf)
+			// check all expected plugin groups are available in the `plugin group search` output from the airgapped repository
+			expectedPluginGroups := []*framework.PluginGroup{{Group: "vmware-tkg/v0.0.1"}}
+			Expect(framework.IsAllPluginGroupsExists(pluginGroups, expectedPluginGroups)).Should(BeTrue(), "all required plugin groups for life cycle tests should exists in plugin group search output")
+
+			// search plugins and make sure correct number of plugins available
+			// check expected plugins are available in the `plugin search` output from the airgapped repository
+			expectedPlugins := pluginsForPG_TKG_001
```

Comment:
Is it possible to avoid hard-coding the plugin information? Instead, can we retrieve the plugin details prior to downloading and utilize them to validate the new airgaped repository?

Reply:
I thought about it and the command that I was missing was `tanzu plugin group get` to get all plugins associated with a plugin group.

Reply:
It could be better to add a TODO statement to the code OR to include some commentary to the story that was created for implementing the command `tanzu plugin group get`.

Reply:
Sure. Added a TODO under `plugins.go` to remove the hardcoding.